components/certificate-mgt/org.wso2.carbon.certificate.mgt.core/src/test/java/org/wso2/carbon/certificate/mgt/core/impl/CertificateManagementServiceImplTests.java

```diff
@@ -0,0 +1,594 @@
+import org.testng.annotations.Test;
+import org.wso2.carbon.certificate.mgt.core.common.BaseDeviceManagementCertificateTest;
+import org.wso2.carbon.certificate.mgt.core.dao.CertificateManagementDAOFactory;
+import org.wso2.carbon.certificate.mgt.core.dto.CAStatus;
+import org.wso2.carbon.certificate.mgt.core.dto.CertificateResponse;
+import org.wso2.carbon.certificate.mgt.core.dto.SCEPResponse;
+import org.wso2.carbon.certificate.mgt.core.exception.CertificateManagementException;
+import org.wso2.carbon.certificate.mgt.core.exception.KeystoreException;
+import org.wso2.carbon.certificate.mgt.core.service.CertificateManagementServiceImpl;
+import org.wso2.carbon.certificate.mgt.core.service.PaginationResult;
+import org.wso2.carbon.certificate.mgt.core.util.CSRGenerator;
+import org.wso2.carbon.certificate.mgt.core.util.CertificateManagementConstants;
+import org.wso2.carbon.certificate.mgt.core.util.DummyCertificate;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.device.mgt.common.DeviceManagementException;
+import org.wso2.carbon.device.mgt.core.config.DeviceConfigurationManager;
+import sun.misc.BASE64Encoder;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.PrivateKey;
+import java.security.Security;
+import java.security.cert.*;
+import java.util.ArrayList;
+import java.util.List;
+
+public class CertificateManagementServiceImplTests extends BaseDeviceManagementCertificateTest {
+
+    private static Log log = LogFactory.getLog(CertificateManagementServiceImplTests.class);
+    private static final String CA_CERT_PEM = "src/test/resources/ca_cert.pem";
+    private static final String RA_CERT_PEM = "src/test/resources/ra_cert.pem";
+    CertificateManagementServiceImpl managementService = null;
+
+    @Test(description = "This test case tests initialization of CertificateManagementServiceImpl instance")
+    public void testGetInstance() {
+        try {
+            CertificateManagementServiceImpl instance = CertificateManagementServiceImpl.getInstance();
+            Assert.assertNotNull(instance);
+            log.info("getInstance Test Successful");
+
+        } catch (NullPointerException e) {
+            log.error("Error while initializing CertificateManagementService", e);
+            Assert.fail();
+        }
+
+
+    }
+
+    @BeforeClass
+    public void initCertificateManagementService() throws DeviceManagementException {
+        //save certificatemanagementservice instance as class variable
+        managementService = CertificateManagementServiceImpl.getInstance();
+        //set Bouncycastle as a provider for testing
+        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
+
+    }
+
+    @Test(description = "This test case tests retrieval of CA Certificate from the keystore")
+    public void testGetCACertificate() {
+        try {
```

Comment:
Please throw all exceptions in the test cases, don't try to catch it, and assert to fail. That's the normal practice.